

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Tools and Databases &mdash; ViromeXplore  documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=5929fcd5"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Troubleshooting" href="troubleshooting.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            ViromeXplore
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow_overview.html">Workflow Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="running_viromexplore.html">Running ViromeXplore</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Tutorial: Analyzing a Virome Dataset with ViromeXplore</a></li>
<li class="toctree-l1"><a class="reference internal" href="troubleshooting.html">Troubleshooting</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tools and Databases</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ViromeXplore</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Tools and Databases</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tools_databases.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tools-and-databases">
<h1>Tools and Databases<a class="headerlink" href="#tools-and-databases" title="Link to this heading">ÔÉÅ</a></h1>
<p>ViromeXplore uses the following versions of the bioinformatics tools:</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 25.0%" />
<col style="width: 10.0%" />
<col style="width: 45.0%" />
<col style="width: 20.0%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Software</p></th>
<th class="head"><p>Version</p></th>
<th class="head"><p>Summary</p></th>
<th class="head"><p>Links</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>VirSorter2</p></td>
<td><p>2.2.3</p></td>
<td><p>Identifies and classifies viral sequences from metagenomic assemblies.</p></td>
<td><p><a class="reference external" href="https://github.com/jiarong/VirSorter2">GitHub</a></p></td>
</tr>
<tr class="row-odd"><td><p>CheckV</p></td>
<td><p>0.8.1</p></td>
<td><p>Quality assessment of viral genomes.</p></td>
<td><p><a class="reference external" href="https://bitbucket.org/berkeleylab/checkv">Bitbucket</a></p></td>
</tr>
<tr class="row-even"><td><p>CD-HIT</p></td>
<td><p>4.8.1</p></td>
<td><p>Clustering tool for reducing sequence redundancy.</p></td>
<td><p><a class="reference external" href="https://sites.google.com/view/cd-hit">CD-HIT</a></p></td>
</tr>
<tr class="row-odd"><td><p>ViromeQC</p></td>
<td><p>1.0.2</p></td>
<td><p>Benchmark and quantify non-viral contamination.</p></td>
<td><p><a class="reference external" href="http://segatalab.cibio.unitn.it/tools/viromeqc/">ViromeQC</a></p></td>
</tr>
<tr class="row-even"><td><p>Kaiju</p></td>
<td><p>1.7.2</p></td>
<td><p>Taxonomic classification of metagenomic reads.</p></td>
<td><p><a class="reference external" href="https://bioinformatics-centre.github.io/kaiju/">Kaiju</a></p></td>
</tr>
<tr class="row-odd"><td><p>Krona</p></td>
<td><p>2.8</p></td>
<td><p>Interactive hierarchical visualization of metagenomic data.</p></td>
<td><p><a class="reference external" href="https://github.com/marbl/Krona/wiki">Github</a></p></td>
</tr>
<tr class="row-even"><td><p>FlashWeave</p></td>
<td><p>1.6</p></td>
<td><p>Infers microbial interaction networks from sequencing data.</p></td>
<td><p><a class="reference external" href="https://github.com/meringlab/FlashWeave">GitHub</a></p></td>
</tr>
<tr class="row-odd"><td><p>TIM</p></td>
<td><p>latest</p></td>
<td><p>Interactions between organisms through phylogeny and co-occurence.</p></td>
<td><p><a class="reference external" href="https://github.com/RomainBlancMathieu/TIM">GitHub</a></p></td>
</tr>
<tr class="row-even"><td><p>fastp</p></td>
<td><p>0.20.1</p></td>
<td><p>FASTQ preprocessing tool (quality filtering, trimming).</p></td>
<td><p><a class="reference external" href="https://github.com/OpenGene/fastp">GitHub</a></p></td>
</tr>
<tr class="row-odd"><td><p>MEGAHIT</p></td>
<td><p>1.2.9</p></td>
<td><p>Ultrafast assembler for large and complex metagenomes.</p></td>
<td><p><a class="reference external" href="https://github.com/voutcn/megahit">GitHub</a></p></td>
</tr>
<tr class="row-even"><td><p>Bowtie2</p></td>
<td><p>2.4.1</p></td>
<td><p>Fast and sensitive short-read alignment.</p></td>
<td><p><a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/">Bowtie2</a></p></td>
</tr>
<tr class="row-odd"><td><p>SAMtools</p></td>
<td><p>1.9</p></td>
<td><p>Utilities for manipulating alignments in SAM/BAM formats.</p></td>
<td><p><a class="reference external" href="http://www.htslib.org/">SAMtools</a></p></td>
</tr>
<tr class="row-even"><td><p>MetaBAT2</p></td>
<td><p>2.15</p></td>
<td><p>Genome binning from metagenomic assemblies.</p></td>
<td><p><a class="reference external" href="https://bitbucket.org/berkeleylab/metabat/src/master/">Bitbucket</a></p></td>
</tr>
<tr class="row-odd"><td><p>COBRA</p></td>
<td><p>1.2.3</p></td>
<td><p>Higher quality viral genomes assembled from metagenomes.</p></td>
<td><p><a class="reference external" href="https://github.com/linxingchen/cobra">GitHub</a></p></td>
</tr>
<tr class="row-even"><td><p>geNomad</p></td>
<td><p>1.8.0</p></td>
<td><p>Virus identification, classification and annotation tool.</p></td>
<td><p><a class="reference external" href="https://portal.nersc.gov/genomad/">geNomad</a></p></td>
</tr>
<tr class="row-odd"><td><p>EggNOG-mapper</p></td>
<td><p>2.1.9</p></td>
<td><p>Functional annotation using orthology assignments.</p></td>
<td><p><a class="reference external" href="https://github.com/eggnogdb/eggnog-mapper/wiki">GitHub</a></p></td>
</tr>
<tr class="row-even"><td><p>VSEARCH</p></td>
<td><p>2.10.4</p></td>
<td><p>Open-source tool for metagenomic clustering, chimera detection, and search.</p></td>
<td><p><a class="reference external" href="https://github.com/torognes/vsearch">GitHub</a></p></td>
</tr>
</tbody>
</table>
<p>ViromeXplore uses the following reference databases.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 25.0%" />
<col style="width: 10.0%" />
<col style="width: 45.0%" />
<col style="width: 20.0%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Database</p></th>
<th class="head"><p>Version</p></th>
<th class="head"><p>Summary</p></th>
<th class="head"><p>Links</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CheckV database</p></td>
<td><p>v1.0</p></td>
<td><p>Database for viral genome quality assessment with CheckV.</p></td>
<td><p><a class="reference external" href="https://portal.nersc.gov/CheckV/checkv-db-v1.0.tar.gz">Download</a></p></td>
</tr>
<tr class="row-odd"><td><p>EggNOG database</p></td>
<td><p>v5.0.2</p></td>
<td><p>Orthology-based functional annotation database used by EggNOG-mapper.</p></td>
<td><p><a class="reference external" href="http://eggnog6.embl.de/download/emapperdb-5.0.2/eggnog_proteins.dmnd.gz">Proteins</a>, <a class="reference external" href="http://eggnog6.embl.de/download/emapperdb-5.0.2/eggnog.db.gz">DB</a>, <a class="reference external" href="http://eggnog6.embl.de/download/emapperdb-5.0.2/eggnog.taxa.tar.gz">Taxa</a></p></td>
</tr>
<tr class="row-even"><td><p>geNomad database</p></td>
<td><p>v1.7</p></td>
<td><p>Reference database for viral discovery and annotation with geNomad.</p></td>
<td><p><a class="reference external" href="https://portal.nersc.gov/genomad/__data__/genomad_db_v1.7.tar.gz">Download</a></p></td>
</tr>
<tr class="row-odd"><td><p>Kaiju virus database</p></td>
<td><p>2024-08-15</p></td>
<td><p>Taxonomic classification database for viral sequences in Kaiju.</p></td>
<td><p><a class="reference external" href="https://kaiju-idx.s3.eu-central-1.amazonaws.com/2024/kaiju_db_viruses_2024-08-15.tgz">Download</a></p></td>
</tr>
<tr class="row-even"><td><p>Virus-Host Database</p></td>
<td><p>latest</p></td>
<td><p>Reference database of known virus-host interactions.</p></td>
<td><p><a class="reference external" href="https://www.genome.jp/ftp/db/virushostdb/virushostdb.formatted.genomic.fna.gz">Download</a></p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Consider citing the tools and databases used in ViromeXplore accordingly.</p>
</div>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="troubleshooting.html" class="btn btn-neutral float-left" title="Troubleshooting" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, rhernandvel.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>