

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Tutorial: Analyzing a Virome Dataset with ViromeXplore &mdash; ViromeXplore  documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=5929fcd5"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Troubleshooting" href="troubleshooting.html" />
    <link rel="prev" title="Workflow Overview" href="workflow_overview.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            ViromeXplore
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="workflow_overview.html">Workflow Overview</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tutorial: Analyzing a Virome Dataset with ViromeXplore</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#quality-control-and-classification">Quality Control and Classification</a></li>
<li class="toctree-l2"><a class="reference internal" href="#viral-assembly">Viral Assembly</a></li>
<li class="toctree-l2"><a class="reference internal" href="#identify-viral-sequences">Identify Viral Sequences</a></li>
<li class="toctree-l2"><a class="reference internal" href="#extend-incomplete-genomes">Extend Incomplete Genomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#taxonomic-annotation">Taxonomic Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#directory-structure">Directory structure</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="troubleshooting.html">Troubleshooting</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ViromeXplore</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Tutorial: Analyzing a Virome Dataset with ViromeXplore</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tutorial-analyzing-a-virome-dataset-with-viromexplore">
<h1>Tutorial: Analyzing a Virome Dataset with ViromeXplore<a class="headerlink" href="#tutorial-analyzing-a-virome-dataset-with-viromexplore" title="Link to this heading"></a></h1>
<p>The following tutorial uses a 10,000 read subset of a virome from the human gut (SRR829034).  The aim of this tutorial is that the user learns to use the workflows contained in the ViromeXplore software. The demo dataset is automatically downloaded with the github repo.</p>
<section id="quality-control-and-classification">
<h2>Quality Control and Classification<a class="headerlink" href="#quality-control-and-classification" title="Link to this heading"></a></h2>
<p>Run the following command to perform quality control and viral classification:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>nextflow<span class="w"> </span>ViromeXplore.nf<span class="w"> </span>--pipeline<span class="w"> </span>qc_classify<span class="w"> </span>--reads<span class="w"> </span><span class="s2">&quot;demo/demo_SRR829034_{1,2}.fastq&quot;</span>
</pre></div>
</div>
<p>The interactive Krona plot can be viewed here:</p>
<p><a class="reference external" href="_static/demo_SRR829034.html">View interactive Krona plot</a></p>
<a class="reference internal image-reference" href="_images/demo_krona.svg"><img alt="Krona plot" src="_images/demo_krona.svg" style="width: 600px;" />
</a>
<p>The ViromeQC results show an enrichment score of <strong>2.26</strong>, indicating that the virome is <strong>2.8×</strong> more enriched than a comparable metagenome.</p>
<p>—</p>
</section>
<section id="viral-assembly">
<h2>Viral Assembly<a class="headerlink" href="#viral-assembly" title="Link to this heading"></a></h2>
<p>Assemble the viral reads:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>nextflow<span class="w"> </span>ViromeXplore.nf<span class="w"> </span>--pipeline<span class="w"> </span>viral_assembly<span class="w"> </span>--reads<span class="w"> </span><span class="s2">&quot;demo/demo_SRR829034_{1,2}.fastq&quot;</span>
</pre></div>
</div>
<p><a class="reference external" href="_static/demo_SRR829034_fastp.html">View interactive  plot</a></p>
<p>The fatsp report shows that the Q30 of most bases was above 30 so no reads were removed.</p>
<a class="reference internal image-reference" href="_images/demo_fastp.png"><img alt="quality plot" src="_images/demo_fastp.png" style="width: 600px;" />
</a>
<p>This workflow also creates an <strong>assembly fasta file</strong> with genomic contigs.
A quick inspection revealed a total of 27 contigs.</p>
</section>
<section id="identify-viral-sequences">
<h2>Identify Viral Sequences<a class="headerlink" href="#identify-viral-sequences" title="Link to this heading"></a></h2>
<p>Run the viral identification workflow on the assembled MAGs:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>nextflow<span class="w"> </span>ViromeXplore.nf<span class="w"> </span>--pipeline<span class="w"> </span>find_viruses<span class="w"> </span>--contigs<span class="w"> </span>results/megahit_output/final.contigs.fa
</pre></div>
</div>
<p><strong>Virsorter2 Results</strong>:
- 3 sequences classified as viral
- 2 = dsDNA phages
- 1 = ssDNA virus</p>
<p>Refer to the following table for more details:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>seqname</p></th>
<th class="head"><p>dsDNAphage</p></th>
<th class="head"><p>NCLDV</p></th>
<th class="head"><p>RNA</p></th>
<th class="head"><p>ssDNA</p></th>
<th class="head"><p>lavidaviridae</p></th>
<th class="head"><p>max_score</p></th>
<th class="head"><p>max_score_group</p></th>
<th class="head"><p>length</p></th>
<th class="head"><p>hallmark</p></th>
<th class="head"><p>viral</p></th>
<th class="head"><p>cellular</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>k77_12</p></td>
<td><p>0.987</p></td>
<td><p>0.260</p></td>
<td><p>0.365</p></td>
<td><p>1.000</p></td>
<td><p>0.267</p></td>
<td><p>1.000</p></td>
<td><p>ssDNA</p></td>
<td><p>2898</p></td>
<td><p>1</p></td>
<td><p>60.000</p></td>
<td><p>0.000</p></td>
</tr>
<tr class="row-odd"><td><p>k77_25</p></td>
<td><p>0.993</p></td>
<td><p>0.253</p></td>
<td><p>0.185</p></td>
<td><p>0.913</p></td>
<td><p>0.280</p></td>
<td><p>0.993</p></td>
<td><p>dsDNAphage</p></td>
<td><p>5029</p></td>
<td><p>4</p></td>
<td><p>62.500</p></td>
<td><p>0.000</p></td>
</tr>
<tr class="row-even"><td><p>k77_26</p></td>
<td><p>1.000</p></td>
<td><p>0.473</p></td>
<td><p>0.120</p></td>
<td><p>0.980</p></td>
<td><p>0.947</p></td>
<td><p>1.000</p></td>
<td><p>dsDNAphage</p></td>
<td><p>6399</p></td>
<td><p>2</p></td>
<td><p>100.000</p></td>
<td><p>0.000</p></td>
</tr>
</tbody>
</table>
<p><strong>Check Viral Genome Quality</strong></p>
<p>Results for CheckV:</p>
<ul class="simple">
<li><p>2 dsDNA viruses: High quality (100% complete)</p></li>
<li><p>ssDNA virus: 55% complete</p></li>
</ul>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>contig_id</p></th>
<th class="head"><p>contig_length</p></th>
<th class="head"><p>proviral_length</p></th>
<th class="head"><p>aai_expected_length</p></th>
<th class="head"><p>aai_completeness</p></th>
<th class="head"><p>aai_confidence</p></th>
<th class="head"><p>aai_error</p></th>
<th class="head"><p>aai_num_hits</p></th>
<th class="head"><p>aai_top_hit</p></th>
<th class="head"><p>aai_id</p></th>
<th class="head"><p>aai_af</p></th>
<th class="head"><p>hmm_completeness_lower</p></th>
<th class="head"><p>hmm_completeness_upper</p></th>
<th class="head"><p>hmm_num_hits</p></th>
<th class="head"><p>kmer_freq</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>k77_12</p></td>
<td><p>2899</p></td>
<td><p>NA</p></td>
<td><p>5203.194186213554</p></td>
<td><p>55.71577566105885</p></td>
<td><p>high</p></td>
<td><p>1.36078157711096</p></td>
<td><p>34</p></td>
<td><p>DTR_883679</p></td>
<td><p>93.88</p></td>
<td><p>92.87</p></td>
<td><p>45.29686796215677</p></td>
<td><p>66.77545783982293</p></td>
<td><p>3</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>k77_25</p></td>
<td><p>5106</p></td>
<td><p>NA</p></td>
<td><p>5006.547801713951</p></td>
<td><p>100.0</p></td>
<td><p>high</p></td>
<td><p>1.48247660798039</p></td>
<td><p>173</p></td>
<td><p>DTR_883655</p></td>
<td><p>100.0</p></td>
<td><p>98.39</p></td>
<td><p>85.48428064735897</p></td>
<td><p>99.99999999999999</p></td>
<td><p>5</p></td>
<td><p>1.01</p></td>
</tr>
<tr class="row-even"><td><p>k77_26</p></td>
<td><p>6476</p></td>
<td><p>NA</p></td>
<td><p>6397.397761926372</p></td>
<td><p>100.0</p></td>
<td><p>high</p></td>
<td><p>1.28739096702271</p></td>
<td><p>62</p></td>
<td><p>DTR_883654</p></td>
<td><p>99.74</p></td>
<td><p>93.64</p></td>
<td><p>95.88582479210984</p></td>
<td><p>100.00000000000001</p></td>
<td><p>3</p></td>
<td><p>1.01</p></td>
</tr>
</tbody>
</table>
<p>—</p>
</section>
<section id="extend-incomplete-genomes">
<h2>Extend Incomplete Genomes<a class="headerlink" href="#extend-incomplete-genomes" title="Link to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>nextflow<span class="w"> </span>ViromeXplore.nf<span class="w"> </span>--pipeline<span class="w"> </span>high_quality_genomes<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--reads<span class="w"> </span><span class="s2">&quot;results/fastp_output/demo_SRR829034_{1,2}.fastp.fq.gz&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--contigs<span class="w"> </span>results/megahit_output/final.contigs.fa<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--viral_contigs<span class="w"> </span>results/checkv_output/viruses.fna
</pre></div>
</div>
<p>Log output shows no extension was performed (expected for demo data):</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>[01/23] Reading contigs and getting the contig end sequences...
[05/23] A total of 3 query contigs were imported.
...
no query was extended, exit! this is normal if you only provide few queries.
</pre></div>
</div>
<p>—</p>
</section>
<section id="taxonomic-annotation">
<h2>Taxonomic Annotation<a class="headerlink" href="#taxonomic-annotation" title="Link to this heading"></a></h2>
<p>Since no high-quality genomes were obtained, we run taxonomy annotation directly on the viral contigs:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>nextflow<span class="w"> </span>ViromeXplore.nf<span class="w"> </span>--pipeline<span class="w"> </span>taxonomy_annotation<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--viral_contigs<span class="w"> </span>results/checkv_output/viruses.fna
</pre></div>
</div>
<p><strong>Taxonomy Results</strong>:
All viral genomese were successfully classified.
- Family: Microviridae</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>seq_name</p></th>
<th class="head"><p>length</p></th>
<th class="head"><p>topology</p></th>
<th class="head"><p>coordinates</p></th>
<th class="head"><p>n_genes</p></th>
<th class="head"><p>genetic_code</p></th>
<th class="head"><p>virus_score</p></th>
<th class="head"><p>fdr</p></th>
<th class="head"><p>n_hallmarks</p></th>
<th class="head"><p>marker_enrichment</p></th>
<th class="head"><p>taxonomy</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>k77_26</p></td>
<td><p>6476</p></td>
<td><p>DTR</p></td>
<td><p>NA</p></td>
<td><p>9</p></td>
<td><p>11</p></td>
<td><p>0.9837</p></td>
<td><p>NA</p></td>
<td><p>2</p></td>
<td><p>14.7373</p></td>
<td><p>Viruses;Monodnaviria;Sangervirae;Phixviricota;Malgrandaviricetes;Petitvirales;Microviridae</p></td>
</tr>
<tr class="row-odd"><td><p>k77_25</p></td>
<td><p>5106</p></td>
<td><p>DTR</p></td>
<td><p>NA</p></td>
<td><p>8</p></td>
<td><p>11</p></td>
<td><p>0.9827</p></td>
<td><p>NA</p></td>
<td><p>3</p></td>
<td><p>11.1680</p></td>
<td><p>Viruses;Monodnaviria;Sangervirae;Phixviricota;Malgrandaviricetes;Petitvirales;Microviridae</p></td>
</tr>
<tr class="row-even"><td><p>k77_12</p></td>
<td><p>2899</p></td>
<td><p>No terminal repeats</p></td>
<td><p>NA</p></td>
<td><p>5</p></td>
<td><p>11</p></td>
<td><p>0.9783</p></td>
<td><p>NA</p></td>
<td><p>1</p></td>
<td><p>4.6856</p></td>
<td><p>Viruses;Monodnaviria;Sangervirae;Phixviricota;Malgrandaviricetes;Petitvirales;Microviridae</p></td>
</tr>
</tbody>
</table>
<p><strong>Functional Annotation</strong>:
- The eggNOG analysis revealed proteins related to:</p>
<blockquote>
<div><ul class="simple">
<li><p>Structural molecule activity</p></li>
<li><p>ATP binding</p></li>
<li><p>Viral process</p></li>
</ul>
</div></blockquote>
<p>—</p>
</section>
<section id="directory-structure">
<h2>Directory structure<a class="headerlink" href="#directory-structure" title="Link to this heading"></a></h2>
<p>Each pipeline step creates its own directory, keeping the workflow organized and reproducible, for example:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>results/
│
├── samtools_output/
├── megahit_output/
├── bowtie_output/
├── vsearch_output/
├── eggnog_mapper_output/
├── checkv_output/
├── viromeQC_output/
├── fastp_output/
├── geNomad_output/
├── virsorter_out/
├── cdhit_output/
├── mapping_summary_output/
├── kaiju_output/
└── cobra_output/
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="workflow_overview.html" class="btn btn-neutral float-left" title="Workflow Overview" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="troubleshooting.html" class="btn btn-neutral float-right" title="Troubleshooting" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, rhernandvel.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>